app/src/main/java/com/zulip/android/activities/MessageAdapter.java

```diff
@@ -1,69 +1,76 @@
-import android.text.Html;
-import android.text.Spanned;
+import android.support.v4.view.ViewCompat;
+import android.support.v7.widget.RecyclerView;
 import android.text.format.DateUtils;
-import android.text.method.LinkMovementMethod;
 import android.util.Log;
+import android.util.TypedValue;
 import android.view.LayoutInflater;
 import android.view.View;
 import android.view.ViewGroup;
-import android.widget.ArrayAdapter;
-import android.widget.ImageView;
-import android.widget.LinearLayout;
-import android.widget.TextView;
 
-import com.zulip.android.util.CustomHtmlToSpannedConverter;
+import com.zulip.android.R;
+import com.zulip.android.ZulipApp;
 import com.zulip.android.filters.NarrowFilterPM;
 import com.zulip.android.filters.NarrowFilterStream;
 import com.zulip.android.filters.NarrowListener;
-import com.zulip.android.R;
-import com.zulip.android.ZulipApp;
 import com.zulip.android.models.Message;
 import com.zulip.android.models.MessageType;
 import com.zulip.android.models.Stream;
-import com.zulip.android.networking.GravatarAsyncFetchTask;
 
-/**
- * Adapter which stores Messages in a view, and generates LinearLayouts for
- * consumption by the ListView which displays the view.
- */
-public class MessageAdapter extends ArrayAdapter<Message> {
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
 
-    private static final String TAG = "MessageAdapter";
-    private static final HTMLSchema schema = new HTMLSchema();
 
+public class MessageAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
+
+    public static final int VIEWTYPE_MESSAGEHEADER = 1;
+    private static final int VIEWTYPE_MESSAGE = 2;
+    private static final int VIEWTYPE_HEADER = 3; //At position 0
+    private static final int VIEWTYPE_FOOTER = 4; //At end position
+
+    private static String privateHuddleText;
+    private List<Object> items;
+    private ZulipApp zulipApp;
+    private Context context;
+    private NarrowListener narrowListener;
+    private static final float HEIGHT_IN_DP = 48;
+    private
     @ColorInt
-    private int mDefaultStreamHeaderColor;
-    @ColorInt
-    private int mDefaultHuddleHeaderColor;
-    @ColorInt
-    private int mDefaultStreamMessageColor;
+    int mDefaultStreamHeaderColor;
+
     @ColorInt
-    private int mDefaultHuddleMessageColor;
+    private int mDefaultPrivateMessageColor;
+    private OnItemClickListener onItemClickListener;
+    private int contextMenuItemSelectedPosition;
+    private boolean startedFromFilter;
+    private View footerView;
+    private View headerView;
+
+    int getContextMenuItemSelectedPosition() {
+        return contextMenuItemSelectedPosition;
+    }
 
-    public MessageAdapter(Context context, List<Message> objects) {
-        super(context, 0, objects);
+    MessageAdapter(List<Message> messageList, final Context context, boolean startedFromFilter) {
+        super();
+        items = new ArrayList<>();
+        setupHeaderAndFooterViews();
```

Comment:
Unless there's a reason for doing this before, it's best practice to initialize fields before calling any methods. (Moving this line below line 69 would fix this).